### ISSUE TITLE
New Custom Fragment syntax doesn't work in Towncrier 21.9.0

### ISSUE THREAD

**Issue by jssullivan** [twisted/towncrier#401]
Towncrier version: 21.9.0

The newer and non deprecated syntax for defining Custom Fragment types isn't working. For a minimum example I just copied and pasted the docs [here](https://towncrier.readthedocs.io/en/latest/configuration.html#custom-fragment-types) and tried running `towncrier create test.feat`, getting the following error
```
> poetry run towncrier create test.feat
Usage: towncrier create [OPTIONS] FILENAME
Try 'towncrier create --help' for help.

Error: Invalid value: Expected filename 'POR-2499.feat' to be of format '{name}.{type}', where '{name}' is an arbitrary slug and '{type}' is one of: feature, bugfix, doc, removal, misc
```
My theory is that this change is slated for the next version of Towncrier. Can the docs be updated and be based on the current latest version instead of the next version?

**Comment by adiroiban** [comment]
Thanks for the report.

Not sure what can be done.

The docs link is for "latest" which is the "trunk" branch.

The Read the docs project is new and we don't have a build for 21.9.0

Here is the failure for 21.9.0 https://readthedocs.org/api/v2/build/16378790.txt

If you can find a way to fix the docs build, we can look into publishing the docs for older version.


21.9.0 has no docs  https://github.com/twisted/towncrier/tree/21.9.0

Fixing it means, backporting the docs to 21.9.0

Right now, I don't have time to backport the Read The docs build process for older towncrier releases.

If you want to create docs for 21.9.0, we can have a PR and I can do a 21.9.1 release.

**Comment by jssullivan** [comment]
Apologies on the long delay.

It looks like latest is now released as 22.8.0. So the docs now correspond to a shipped version of towncrier.

I will close this for now. My confusion was during integration, reading that the old fragment is deprecated, but the new syntax wasn't working (as it wasn't released yet).

This new release resolves this issue :)
